mailpile/search.py

```diff
@@ -605,6 +605,9 @@ def _real_scan_one(self, session,
             last_date = long(msg_info[self.MSG_DATE], 36)
             added += 1
 
+        PluginManager.trigger(
```

Comment:
I don't think this is the right place for it; in this place we may be "lazy" which means we're adding metadata about the message to our index but not actually processing the message body. You want the message body to be available when this takes place; that basically means you want to look for where the keyword extractors are invoked. There is already a hook there for plugins to look at incoming mail, so I'm not sure we need a new one for this.